vowpalwabbit/topk.cc

```diff
@@ -17,7 +17,7 @@ using scored_example = pair<float, v_array<char>>;
 
 struct compare_scored_examples
 {
-  bool operator()(scored_example const& a, scored_example const& b) const { return a.first > b.first; }
+  constexpr bool operator()(scored_example const& a, scored_example const& b) const { return a.first > b.first; }
```

Comment:
We have places where we use "function structs" and places where we use bare function pointers. Should we have an issue about deciding on one of these two patterns and unifying?

Reply:
The functor used here is a requirement I believe for where it is used? Ideally I'd like to see *function* arguments templated so any form of callable can be supplied (functor, lambda, function pointer, std::function)